stanford_events_importer.install

```diff
@@ -760,3 +762,72 @@ function stanford_events_importer_update_7304(&$sandbox) {
     field_cache_clear();
   }
 }
+
+/**
+ * Update variables to point to events-legacy.stanford.edu.
+ */
+function stanford_events_importer_update_7305(&$sandbox) {
+  $category_source = variable_get('stanford_events_category_source');
+  if ($category_source == "http://events.stanford.edu/xml/drupal/v2.php?category-list") {
+    variable_set('stanford_events_category_source', 'https://events-legacy.stanford.edu/xml/drupal/v2.php?category-list');
+    watchdog('stanford_events_importer', 'Stanford Events Importer category list updated to pull from events-legacy.stanford.edu');
+  }
+  $organization_source = variable_get('stanford_events_organization_source');
+  if ($organization_source == 'http://events.stanford.edu/xml/drupal/v2.php?organization-list') {
+    variable_set('stanford_events_organization_source', 'https://events-legacy.stanford.edu/xml/drupal/v2.php?organization-list');
+    watchdog('stanford_events_importer', 'Stanford Events Importer organization list updated to pull from events-legacy.stanford.edu');
+  }
+}
+
+/**
+ * Update feeds_source config to point to events-legacy.stanford.edu.
+ */
+function stanford_events_importer_update_7306(&$sandbox) {
+  // See also stanford_events_importer_update_7201().
+  $configs = db_select('feeds_source', 'fs')
+    ->fields('fs',
+      array(
+        'feed_nid',
+        'config',
+        'source',
+      ))
+    ->condition('id', 'stanford_event_importer')
+    ->execute();
+
+  // Loop through each of the configured nodes and update the URL.
+  while ($values = $configs->fetchAssoc()) {
+    $nid = $values['feed_nid'];
+    $importer_config = unserialize($values['config']);
+    $source = $values['source'];
+    $source = str_replace('events.stanford.edu/xml/drupal/v2.php', 'events-legacy.stanford.edu/xml/drupal/v2.php', $source);
+    $importer_config['FeedsHTTPFetcher']['source'] = $source;
+
+    db_update('feeds_source')
+      ->fields(array(
+        'config' => serialize($importer_config),
+        'source' => $source,
+      ))
+      ->condition('feed_nid', $nid)
+      ->execute();
+  }
+  watchdog('stanford_events_importer', 'Stanford Events Importer feeds updated to pull from events-legacy.stanford.edu');
+
+}
+
+/**
+ * Update field_stanford_event_add_to_cal.
+ */
+function stanford_events_importer_update_7307(&$sandbox) {
+  $query = new EntityFieldQuery();
+  $result = $query->entityCondition('entity_type', 'node')
+    ->entityCondition('bundle', 'stanford_event')
```

Comment:
How about also a condition for the field you're updating? That'll reduce the number of nodes.

Reply:
```
  ->fieldCondition('field_stanford_event_add_to_cal', 'field_stanford_event_add_to_cal_url', '%events.stanford.edu%', 'LIKE')
```

Generates the error:
```
 ID         :  727                                                                                                                                                                                                                              
 Date       :  27/Jan 12:52                                                                                                                                                                                                                     
 Type       :  php                                                                                                                                                                                                                              
 Severity   :  error                                                                                                                                                                                                                            
 Message    :  PDOException: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'field_data_field_stanford_event_add_to_cal0.field_stanford_event_add_to_cal_field_stanford_event_add_to_cal_url' in 'where clause': SELECT                 
               field_data_field_stanford_event_add_to_cal0.entity_type AS entity_type, field_data_field_stanford_event_add_to_cal0.entity_id AS entity_id, field_data_field_stanford_event_add_to_cal0.revision_id AS revision_id,              
               field_data_field_stanford_event_add_to_cal0.bundle AS bundle                                                                                                                                                                     
               FROM                                                                                                                                                                                                                             
               {field_data_field_stanford_event_add_to_cal} field_data_field_stanford_event_add_to_cal0                                                                                                                                         
               WHERE  (field_data_field_stanford_event_add_to_cal0.field_stanford_event_add_to_cal_field_stanford_event_add_to_cal_url LIKE :db_condition_placeholder_0 ESCAPE '\\') AND (field_data_field_stanford_event_add_to_cal0.deleted = 
               :db_condition_placeholder_1) AND (field_data_field_stanford_event_add_to_cal0.entity_type = :db_condition_placeholder_2) AND (field_data_field_stanford_event_add_to_cal0.bundle = :db_condition_placeholder_3) ; Array          
               (                                                                                                                                                                                                                                
                   [:db_condition_placeholder_0] => %events.stanford.edu%                                                                                                                                                                       
                   [:db_condition_placeholder_1] => 0                                                                                                                                                                                           
                   [:db_condition_placeholder_2] => node                                                                                                                                                                                        
                   [:db_condition_placeholder_3] => stanford_event                                                                                                                                                                              
               )                                                                                                                                                                                                                                
                in field_sql_storage_field_storage_query() (line 748 of /app/docroot/modules/field/modules/field_sql_storage/field_sql_storage.module).  
```

Need to figure out what to use for `[:db_condition_placeholder_1]`

Troubleshooting in between meetings, so making notes as I go here, intermittently.

Reply:
Oh, it doesn't like `LIKE` for the operator. OK.

Reply:
I was wondering about that `condition('source', '%events.stanford.edu%', 'LIKE')` is the D9 way. I'm not sure how to do it in the D7

Reply:
It apparently should be `CONTAINS`, but even with that, it's populating `[:db_condition_placeholder_1]` with `0`, and appending that to the end of the field name (e.g., `SELECT field_data_field_stanford_event_add_to_cal0.entity_type ...`)

Reply:
There's a `db_condition_placeholder` that I'm missing...

Reply:
I'm going to time-box this one as well. I'll bang on it a little more tonight, but if I can't chase down the fix, it's probably not worth our time to try to optimize, especially for a one-time operation.

Reply:
yeah batching the update is going to be more efficient use of time. Let's at least focus on that.